examples/uniform_heat_equation.cpp

```diff
@@ -0,0 +1,313 @@
+// Copyright (C) The DDC development team, see COPYRIGHT.md file
+//
+// SPDX-License-Identifier: MIT
+
+//! [includes]
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+
+#include <ddc/ddc.hpp>
+//! [includes]
+
+
+//! [X-dimension]
+struct X;
+//! [X-dimension]
+
+//! [X-discretization]
+struct DDimX : ddc::UniformPointSampling<X>
+{
+};
+//! [X-discretization]
+
+//! [Y-space]
+struct Y;
+struct DDimY : ddc::UniformPointSampling<Y>
+{
+};
+//! [Y-space]
+
+//! [time-space]
+struct T;
+struct DDimT : ddc::UniformPointSampling<T>
+{
+};
+//! [time-space]
+
+//! [display]
+/** A function to pretty print the temperature
+ * @tparam ChunkType The type of chunk span. This way the template parameters are avoided,
+ *                   should be deduced by the compiler.
+ * @param time The time at which the output is made.
+ * @param temp The temperature at this time-step.
+ */
+template <class ChunkType>
+void display(double time, ChunkType temp)
+{
+    double const mean_temp = ddc::transform_reduce(
+                                     temp.domain(),
+                                     0.,
+                                     ddc::reducer::sum<double>(),
+                                     temp)
+                             / temp.domain().size();
+    std::cout << std::fixed << std::setprecision(3);
+    std::cout << "At t = " << time << ",\n";
+    std::cout << "  * mean temperature  = " << mean_temp << "\n";
+    ddc::ChunkSpan temp_slice
+            = temp[ddc::get_domain<DDimY>(temp).front()
+                   + ddc::get_domain<DDimY>(temp).size() / 2];
+    std::cout << "  * temperature[y:"
+              << ddc::get_domain<DDimY>(temp).size() / 2 << "] = {";
+    ddc::for_each(
+            ddc::get_domain<DDimX>(temp),
+            [=](ddc::DiscreteElement<DDimX> const ix) {
+                std::cout << std::setw(6) << temp_slice(ix);
+            });
+    std::cout << " }" << std::endl;
+
+#ifdef DDC_BUILD_PDI_WRAPPER
+    ddc::PdiEvent("display")
+            .with("temp", temp)
+            .and_with("mean_temp", mean_temp)
+            .and_with("temp_slice", temp_slice);
+#endif
+}
+//! [display]
+
+
+//! [main-start]
+int main(int argc, char** argv)
+{
+#ifdef DDC_BUILD_PDI_WRAPPER
+    auto pdi_conf = PC_parse_string("");
+    PDI_init(pdi_conf);
+#endif
+    Kokkos::ScopeGuard const kokkos_scope(argc, argv);
+    ddc::ScopeGuard const ddc_scope(argc, argv);
+
+
+    //! [parameters]
+    double const x_start = -1.;
+    double const x_end = 1.;
+    size_t const nb_x_points = 10;
+    double const kx = .01;
+    double const y_start = -1.;
+    double const y_end = 1.;
+    size_t const nb_y_points = 100;
+    double const ky = .002;
+    double const start_time = 0.;
+    double const end_time = 10.;
+    ptrdiff_t const t_output_period = 10;
+    //! [parameters]
+
+    //! [main-start]
+    //! [X-parameters]
+    ddc::DiscreteVector<DDimX> static gwx(1);
```

Comment:
```suggestion
    ddc::DiscreteVector<DDimX> const gwx(1);
```

Reply:
Why was it put in static initially ?

Reply:
`static constexpr` ensures that the object is constructed at compile time. `constexpr` only is not enough to ensure it if the object is not used in a constant expression.